bots/make-checkout

```diff
@@ -46,54 +48,36 @@ def main():
 
     api = github.GitHub(repo=opts.repo)
 
-    def execute(*args):
+    def execute(*args, cwd=BASE, error_on_fail=True):
+        output = None
         if opts.verbose:
             sys.stderr.write("+ " + " ".join(args) + "\n")
         try:
-            output = subprocess.check_output(args, cwd=BASE, universal_newlines=True)
+            output = subprocess.check_output(args, cwd=cwd, universal_newlines=True)
         except subprocess.CalledProcessError as ex:
-            sys.exit(ex.returncode)
-        if opts.verbose and output:
-            sys.stderr.write("> " + output + "\n")
+            sys.exit(ex.returncode if error_on_fail else 0)
+        finally:
+            if opts.verbose and output:
+                sys.stderr.write("> " + output + "\n")
         return output
 
-    # do this before fetching the actual test repo, so that FETCH_HEAD always points to that at the end
-    if opts.bots_ref:
-        execute("git", "fetch", "origin", opts.bots_ref)
-        bots_ref = execute("git", "rev-parse", "FETCH_HEAD").strip()
-    else:
-        bots_ref = "origin/master"
-
-    # Testing external project, firstly we needs to add remote
-    if api.repo != "cockpit-project/cockpit":
-        if "test\n" in subprocess.check_output([ "git", "remote" ], universal_newlines=True):
-            execute("git", "remote", "remove", "test")
-        execute("git", "remote", "add", "test", "https://github.com/" + api.repo)
+    if os.path.exists(TARGET_DIR):
+        shutil.rmtree(TARGET_DIR)
 
-    # if the pr was updated while the command was in the queue, the revision
-    # won't be tied to the ref anymore, and it won't exist locally, so exit
-    # silently
-    output = None
-    try:
-        args = ["git", "fetch", "test" if api.repo != "cockpit-project/cockpit" else "origin", opts.ref]
-        if opts.verbose:
-            sys.stderr.write("+ {0}\n".format(" ".join(args)))
-        output = subprocess.check_output(args, cwd=BASE, universal_newlines=True)
-        args = ["git", "checkout", "--detach", opts.revision]
-        if opts.verbose:
-            sys.stderr.write("+ {0}\n".format(" ".join(args)))
-        output = subprocess.check_output(args, cwd=BASE, universal_newlines=True)
-    except subprocess.CalledProcessError:
-        return 0
-    finally:
-        if opts.verbose and output:
-            sys.stderr.write("> " + output + "\n")
+    cache = os.getenv('XDG_CACHE_HOME', "")
+    if cache:
```

Comment:
Wouldn't `reference-if-able` take care of this? That when the cache does not exists then it just happily (with warning) clones  it from remote?

Also how do we plan to keep the cache up to date?

Reply:
Oh yea, I can eliminate that.

The cache is actually the directories cockpituous keeps up to date in cockpit-tasks, which are updated each time any task is performed.

Reply:
so this clone in reality should only cause the network traffic of the few commits that belong to a PR